swagger2.0/oic.r.rule.expression.swagger.json

```diff
@@ -0,0 +1,156 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "OCF Rule Expression",
+    "version": "20200114",
+    "license": {
+      "name": "OCF Data Model License",
+      "url": "https://github.com/openconnectivityfoundation/core/blob/e28a9e0a92e17042ba3e83661e4c0fbce8bdc4ba/LICENSE.md",
+      "x-copyright": "copyright 2020 Open Connectivity Foundation, Inc. All rights reserved."
+    },
+    "termsOfService": "https://openconnectivityfoundation.github.io/core/DISCLAIMER.md"
+  },
+  "schemes": ["http"],
+  "consumes": ["application/json"],
+  "produces": ["application/json"],
+  "paths": {
+    "/RuleExpressionResURI": {
+      "get": {
+        "description": "Expression for the Rule that defines the Rule logic in terms of the defined Rule Inputs, and which evaluates to a boolean value, for which \"true\" means that the Rule has been triggered.",
+        "parameters": [
+          {"$ref": "#/parameters/interface-all"}
+        ],
+        "responses": {
+          "200": {
+            "description" : "Retrieves the Rule expression.",
+            "x-example":
+            {
+              "rt": ["oic.r.rule.expression"],
+              "if": ["oic.if.rw", "oic.if.baseline"],
+              "rule": "(mytemperature:temperature >= 25)",
+              "ruleresult": false,
+              "ruleenable": true,
+              "actionenable": true
+            },
+            "schema": { "$ref": "#/definitions/ruleexpression" }
+          }
+        }
+      },
+      "post": {
+        "description": "",
+        "parameters": [
+          {"$ref": "#/parameters/interface-rw"},
+          {
+            "name": "body",
+            "in": "body",
+            "required": true,
+            "schema": { "$ref": "#/definitions/ruleexpression-update" },
+            "x-example":
+              {
+                "ruleenable": true,
+                "actionenable": false,
+                "ruleresult": false
+              }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description" : "",
+            "x-example": {
+              "rule": "(mytemperature:temperature >= 25)",
```

Comment:
Why is the Rule Expression in parenthesis?  The examples in the CR don't have them.

Reply:
It's valid as per the BNF. It doesn't matter.